### ISSUE TITLE
Project "Ion-Prop": A Next-Generation Ionospheric Forecasting Model for Radio Communications

### ISSUE THREAD

**Issue by MartialTerran** [NASA-IMPACT/Surya#33]
The goals for this addition to the Surya model family was inspired by a conversation between Martial Terran and a HAM Radio enthusiast at the Harwich Cranberry Festival on Saturday September 13, 2025 in Harwich, Mass (Cape Cod).  Martial explained the capabilities of the new NASA-Surya model and described some of Martial's upgrade proposals (issues).  The Ham Operator explained to Martial the varied effect of sun spots upon terrestrial (atmospherically-bounced signals) and extraterrestrial (point to point) HAM Radio communications and he agreed enthusiastically with Martial that an ionization-predicting model would be very helpful.

---

### **Project "Ion-Prop": A Next-Generation Ionospheric Forecasting Model for Radio Communications**

**Labels:** `enhancement`, `new-model`, `community-collaboration`, `help-wanted`

**Opened by:** "Martial Terran"  mrferran1970@gmail.com  https://huggingface.co/MartialTerran
**Date:** September 15, 2025

### Summary

Following a highly successful [AI-Simulated] joint conference between NASA/Surya developers and amateur radio operators [copied below], we are thrilled to propose **Project "Ion-Prop"**: a new, specialized predictive model designed to translate Surya's solar insights into actionable, real-time ionospheric forecasts for HAM radio, military, and commercial communications.

The vision is to create a system that can provide a hyper-local, on-demand "space weather report" for any two points on Earth, answering the critical question: "What will the sun's activity *do* to my signal path?" This project will leverage Surya's foundational vision model and combine it with multiple real-time data streams to predict radio propagation conditions with unprecedented accuracy and user-friendliness.

### The User Story

As a HAM radio operator, I want to know the best time and frequency to contact a specific region or another operator. I can see the sunspots on a solar dashboard, but I don't know exactly when or how they will affect my specific communication path. I need a tool that can:
1.  Predict the probability of a successful contact between my location (Grid Square A) and a target location (Grid Square B).
2.  Provide this prediction in an easy-to-understand, natural language report.
3.  Offer hour-by-hour forecasts for the near future.
4.  Visually show me how the ionosphere is behaving over my region of interest.

### Proposed Solution & Development Roadmap

We propose a phased development plan to build the Ion-Prop model, starting with a text-based forecasting service and evolving to include advanced visual generative capabilities.

---

### **Phase 1: The Foundational Model (Ion-Prop V1)**

The initial version will focus on creating the core predictive engine and a user-friendly, text-based reporting system.

#### **1. Multi-Faceted Dataset Integration**

The model will be trained on a diverse and comprehensive dataset, connecting solar drivers to terrestrial impacts:

*   **Solar Input:**
    *   **Surya Embeddings:** We will use the rich, high-dimensional vector representations of solar activity generated by the core Surya model from SDO imagery. This is our "eye on the sun."
*   **Interplanetary Data:**
    *   **Real-time Solar Wind:** Data from sources like the DSCOVR satellite (speed, density, Bz, etc.) to model the solar wind's impact on the magnetosphere.
*   **Ionospheric Ground Truth:**
    *   **Total Electron Content (TEC):** Global and regional TEC maps will serve as a direct measure of ionospheric ionization.
    *   **Citizen Science Propagation Data:** This is our "secret sauce" for path-specific predictions. We will ingest massive, real-time datasets from:
        *   **WSPRNet:** For weak signal propagation paths and signal-to-noise ratios.
        *   **Reverse Beacon Network (RBN):** For real-world CW/RTTY contact data.
        *   **PSKReporter:** For digital mode contact data.

#### **2. Model Architecture**

*   **Core Predictive Engine:** A multi-input transformer or a similar sequence-to-sequence model capable of integrating these disparate data types. The model's attention mechanisms will learn the complex time-delayed relationships between a solar event (from Surya) and the resulting propagation path success (from WSPR/RBN).
*   **Classification Method:** The primary output will be a probabilistic forecast. For a given path (Grid A -> Grid B), time, and frequency band, the model will output:
    *   `P(contact)`: The probability of a successful contact (e.g., 0.85).
    *   `Predicted SNR`: The likely signal-to-noise ratio.
    *   `Dominant Layer`: The predicted primary propagation layer (e.g., F2, E-skip).

#### **3. Output & Delivery Modes**

*   **On-Demand "Ionospheric Weather" Reports (Text):**
    *   Users can query an API or a web interface with their location, a target location, and a time window.
    *   The system will return a human-readable report.
    *   **Example Output:**
        ```
        Report for path: London (JO01) to Tokyo (PM95)
        Next 12 Hours Forecast (Starting 16-Sep-2025 00:00Z):
        - 10m Band: Poor. High D-layer absorption predicted until 07:00Z.
        - 15m Band: Good opening expected between 08:00Z and 11:00Z.
        - 20m Band: **BEST OPTION**. Path predicted to be open 06:00Z - 14:00Z.
                   Peak probability of contact (85%) at 09:30Z via F2 layer.
        ```

*   **Interactive LLM Explanations:**
    *   We will integrate a fine-tuned Language Model to act as a conversational interface. This LLM will translate the model's raw probabilistic output and provide deeper insights.
    *   **Example Interaction:**
        > **User:** "I want to establish a path from London to Tokyo, tell me the likely conditions for that specific path."
        >
        > **Ion-Prop LLM:** "For the path from London to Tokyo, your best bet over the next 12 hours is the 20-meter band, which is predicted to open around 06:00 UTC. Conditions should peak with an 85% probability of a successful contact around 09:30 UTC, thanks to a strong enhancement of the F2 layer. Would you like me to check conditions on the 15-meter band as well?"

---

### **Phase 2: Visual Forecasting (Ion-Prop V2)**

Building on the success of V1, the second phase will introduce a generative visual model to create intuitive, dynamic maps of ionospheric conditions.

*   **Generative Model Architecture:**
    *   We will develop a **geographically-aware diffusion model** or a similar generative AI (GAN, etc.). This model will be trained on historical TEC maps and conditioned on the predictive output of the Ion-Prop V1 engine.

*   **Output Mode: Dynamic Heatmaps:**
    *   The model will generate animated GIFS or interactive map overlays showing predicted ionospheric conditions for a user-specified region.
    *   This will provide an at-a-glance, visual understanding of propagation.
    *   **Visual Legend:**
        *   **Deep Reds:** Indicate strong D-layer absorption (HF blackout).
        *   **Vibrant Greens/Yellows:** Indicate high Maximum Usable Frequency (MUF) over the F2 layer (fantastic DX opportunities).
        *   **Pulsing Blues:** Indicate sporadic-E cloud formation.
    *   This allows a user to visually see where the "good" and "bad" areas of the ionosphere are for HF communication in near-real time.

### Call for Collaboration

This is an ambitious project that sits at the intersection of heliophysics, AI, and citizen science. We openly invite collaboration from the amateur radio community, data scientists, and anyone passionate about this challenge. Your data, expertise, and feedback are critical to making Project Ion-Prop a success. Let's build the future of space weather forecasting together

---

## Dawn of a New Predictive Era: HAM Radio Operators and NASA Experts Convene to Chart the Future of Ionospheric Forecasting

**Proceedings of the Inaugural Conference on AI-Enhanced Ionospheric Prediction for Amateur Radio**

**Date:** September 15, 2025

**Attendees:**

*   **Dr. Evelyn Reed,** NASA Heliophysics Division, Surya Project Scientist
*   **Mr. Kenji Tanaka,** Lead Machine Learning Engineer, NASA IMPACT Team
*   **Ms. Fatima Al-Jamil,** ML Researcher, IBM (Surya Collaboration)
*   **Mr. John "Skip" Miller (W1XYZ),** American Radio Relay League (ARRL) Representative
*   **Ms. Maria Hernandez (XE2ABC),** Radio Amateurs of Canada (RAC) Representative
*   **Mr. David Chen (GMDXYZ),** Radio Society of Great Britain (RSGB) Representative
*   A global audience of HAM radio operators and space weather enthusiasts.

**(The conference hall buzzes with anticipation. A large screen displays the NASA and Surya logos. Dr. Evelyn Reed takes the podium.)**

**Dr. Evelyn Reed:** Good morning, everyone. On behalf of NASA and our partners at IBM, I'm thrilled to welcome you to this landmark event. We're here today to bridge two worlds: the cutting-edge of heliophysics and artificial intelligence, and the vibrant, global community of amateur radio. Our focus is the Surya model, a foundational AI for heliophysics, and its potential to revolutionize how we understand and predict the ionosphere for radio communications.

Surya, trained on nine years of high-resolution images from the Solar Dynamics Observatory, has shown incredible promise in forecasting solar activity like flares, solar wind, and coronal mass ejections. It's a powerful tool, a "weather forecaster" for the sun.

But we know that a solar forecast is not a radio forecast. That's why we're here to talk with you, the experts in making contacts under all sorts of conditions. We want to work together to build something new, something that directly serves your needs.

**(Dr. Reed gestures to Mr. Tanaka.)**

**Mr. Kenji Tanaka:** Thanks, Evelyn. To get us started, let's talk about what Surya can do right now. Surya's "classification heads" are essentially its specialized outputs. Currently, we have heads for:

*   **Solar Flare Prediction:** A binary classification of whether a solar flare is likely to occur.
*   **Solar Wind Speed Forecasting:** Predicting the speed of the solar wind.
*   **Active Region Segmentation:** Identifying and tracking active regions on the sun.

These are fantastic for giving us a general idea of solar activity. However, for a HAM radio operator, knowing a flare is coming is only half the story. You need to know what that flare will *do* to the ionosphere above you. That's the missing piece.

**(Mr. Miller, W1XYZ, steps up to the microphone.)**

**Mr. John "Skip" Miller (W1XYZ):** Exactly, Kenji. We see a big sunspot group, and we know conditions are likely to get interesting. But the "when" and "where" of the impact are what matter for that DX contact. Will the D-layer be so energized that it absorbs all our signals? Or will the F2 layer be enhanced for a fantastic opening on 10 meters? That's the actionable intelligence we need. A general "flare likely" prediction is a good start, but it's not the targeted forecast we dream of.

**Dr. Reed:** This is precisely the conversation we came here to have. We believe that by working with the amateur radio community, we can develop a new set of classification heads for Surya, and perhaps even a new, specialized model, that can provide the targeted predictions you need. Let's call this new model, for now, the "Ionospheric Propagation" or "Ion-Prop" model.

**(A slide appears on the screen titled "Project Ion-Prop: A New Predictive Model")**

**Ms. Fatima Al-Jamil:** To build Ion-Prop, we need to train it on the right data. We need to teach it to connect the dots between what Surya sees on the sun and what you experience on the airwaves. This will require a multi-faceted dataset:

**Input Data Types for the Ion-Prop Model:**

1.  **Solar Imagery and Surya Embeddings:** We'll use the core Surya model as our "eyes on the sun." Its internal representations of solar images will be a key input. This gives us a deep, nuanced understanding of the solar state.
2.  **Solar Wind Data:** Real-time data from satellites like the Deep Space Climate Observatory (DSCOVR) on solar wind speed, density, and the interplanetary magnetic field (IMF) are crucial.
3.  **Ionospheric Data:** This is where we need to get creative and collaborative.
    *   **Total Electron Content (TEC):** Data from global navigation satellite system (GNSS) receivers, which measure the total number of electrons in a column between the satellite and a ground station.
    *   **Ionosonde Data:** Ground-based ionosondes provide direct measurements of the height and density of the ionosphere's layers.
4.  **Amateur Radio Propagation Data (The "Secret Sauce"):**
    *   **Weak Signal Propagation Reporter (WSPRNet):** This is an invaluable resource. The global network of WSPR stations provides real-time, quantitative data on signal propagation paths and signal-to-noise ratios on various bands.
    *   **Reverse Beacon Network (RBN):** Provides real-time spotting of CW and RTTY signals, giving us a massive dataset of successful communication paths.
    *   **PSKReporter:** Similar to the RBN, but for digital modes like PSK31.

**(Ms. Hernandez, XE2ABC, raises her hand.)**

**Ms. Maria Hernandez (XE2ABC):** This is very exciting. The amateur community has been collecting this data for years. So, to be clear, the idea is to feed all of this data—the sun pictures, the solar wind, the TEC data, and our on-the-air reports—into the new Ion-Prop model?

**Mr. Kenji Tanaka:** Precisely. The model will learn the complex, non-linear relationships between a specific solar event and the resulting ionospheric response, as measured by TEC and confirmed by your real-world propagation data.

**Mr. David Chen (GMDXYZ):** That brings me to a key question: can this new model predict ionization at a specific region of interest? For instance, if I want to establish a path from London to Tokyo, can it tell me the likely conditions for that specific path?

**Dr. Reed:** That is the ultimate goal, and we believe it is achievable. By incorporating the location data of the transmitting and receiving stations from WSPRNet, the RBN, and PSKReporter, the Ion-Prop model can be trained to make **region-specific and path-specific predictions.** It won't just say "the 15-meter band is open." It will be able to predict the probability of a successful contact between two specific grid squares, based on the current and predicted solar-ionospheric conditions.

**Conference Resolutions and Path Forward:**

The conference concluded with a unanimous agreement to move forward with the "Ion-Prop" project. The key resolutions were:

1.  **A New Model is Needed:** While Surya is a powerful foundation, a new, specialized model ("Ion-Prop") is required to translate solar observations into actionable ionospheric predictions for the amateur radio community and other stakeholders.
2.  **Data is Key:** The success of Ion-Prop will depend on the integration of diverse datasets: solar imagery from SDO, real-time solar wind data, ionospheric measurements like TEC, and large-scale amateur radio propagation data.
3.  **Citizen Science Collaboration:** The amateur radio community will play a vital role by continuing to contribute to and expand networks like WSPRNet, the RBN, and PSKReporter. A new initiative will be launched to streamline the collection and formatting of this data for use in the Ion-Prop model.
4.  **Region-Specific Predictions are Feasible:** The conference participants agreed that with sufficient location-based propagation data, the Ion-Prop model can be trained to provide predictions for specific geographic regions and communication paths.
5.  **Open Source and Open Data:** The new Ion-Prop model, like Surya, will be developed as an open-source project. The curated datasets used for its training will also be made publicly available to foster further research and innovation.

**(Dr. Reed returns to the podium for a final statement.)**

**Dr. Evelyn Reed:** Today, we have charted a course for a new era of predictive science, one where the power of AI is combined with the passion and expertise of a global community. The road ahead will be challenging, but the potential rewards are immense: a deeper understanding of our star's influence on our planet, and a powerful new tool to enhance and protect the communications systems that connect us all. Let's get to work.

---
Martial's Prompt for Above Issue Statement:

Now, the conference will publish an "issue" to be added to the Surya Github outlining NASA's plan to construct a model to predict and report (to individual HAM operators who specify their latitude/longitude) (trained to make region-specific and path-specific predictions) so they can know what that flare will do to the ionosphere in the region of interest to them in near-real time, or as far as the model can predicted (outputting a hour-by-hour local HAM weather prediction (for between two operators etc, or regions etc.)   An additional language model will be added to explain in plain English what the vision model's prediction is.   It will be able to predict the probability of a successful contact between two specific grid squares, based on the current and predicted solar-ionospheric conditions.   The first version of the new model will be developed for producing human-readable text reports, but will be provisioned so that in later versions a diffusion model can generate geographically accurate heat maps (GIFS) of ionization in a region of interest on Earth (visually showing where "the D-layer be so energized that it absorbs all our signals" visually showing where "the F2 layer be enhanced for a fantastic opening on 10 meters").  The NASA "issue" plan will outline this ultimate development path.

Write a upbeat Github "issue" title to introduce the "Ionospheric Propagation" or "Ion-Prop" model, and then proceed to draft the detailed and complete NASA conference's PLAN in Github "issue" format (markdown), including: a description of the require a multi-faceted dataset, model architectures, classification methods, planned output modes (on-demand weather report texts) (plus interactive LLM explanations: "I want to establish a path from London to Tokyo, tell me the likely conditions for that specific path, times, etc."), push-mode 2D diffusion maps etc).
